### ISSUE TITLE
feature discussion: define a Model that can be composed of multiple tables joined together

### ISSUE THREAD

**Issue by tjwebb** [balderdashy/waterline#589]
I'd like to define a waterline model `M` that is the representation of a join between tables `t1` and `t2`. Is this possible? e.g.

```js
{
  join: {
    t1: 'id',
    t2: 't1_id'
  },    
  attributes: {
    foo: {
      columnName: 't1.foo'
    },
    bar: {
      columnName: 't2.bar'
    }
  }
}
```
or something, etc etc making stuff up

**Comment by devinivy** [comment]
The closest thing to docs here are my comment on #705 https://github.com/balderdashy/waterline/issues/705#issuecomment-60945411

**Comment by sailsbot** [comment]
Thanks for posting, @tjwebb.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
